File: ReactAndroid/src/main/java/com/facebook/yoga/YogaMeasureOutput.java
/*
 * Copyright (c) 2014-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.yoga;

/**
 * Helpers for building measure output value.
 */
public class YogaMeasureOutput {

  public static long make(float width, float height) {
    final int wBits = Float.floatToRawIntBits(width);
    final int hBits = Float.floatToRawIntBits(height);
    return ((long) wBits) << 32 | ((long) hBits);
  }

  public static long make(int width, int height) {
    return make((float) width, (float) height);
  }

  public static float getWidth(long measureOutput) {
    return Float.intBitsToFloat((int) (0xFFFFFFFF & (measureOutput >> 32)));
  }

  public static float getHeight(long measureOutput) {
    return Float.intBitsToFloat((int) (0xFFFFFFFF & measureOutput));
  }
}

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/yoga/YogaMeasureOutput.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.yoga

/** Helpers for building measure output value. */
public object YogaMeasureOutput {
  @JvmStatic
  public fun make(width: Float, height: Float): Long {
    val wBits = java.lang.Float.floatToRawIntBits(width)
    val hBits = java.lang.Float.floatToRawIntBits(height)
    return (wBits.toLong()) shl 32 or (hBits.toLong())
  }

  @JvmStatic
  public fun make(width: Int, height: Int): Long = make(width.toFloat(), height.toFloat())

  @JvmStatic
  public fun getWidth(measureOutput: Long): Float =
      java.lang.Float.intBitsToFloat((0xFFFFFFFFL and (measureOutput shr 32)).toInt())

  @JvmStatic
  public fun getHeight(measureOutput: Long): Float =
      java.lang.Float.intBitsToFloat((0xFFFFFFFFL and measureOutput).toInt())
}
